docs/examples/EnsembleFilterExample.py

```diff
@@ -0,0 +1,180 @@
+        timestamp=start_time + timedelta(seconds=k)))
+
+# %%
+
+plotter = Plotterly()
+plotter.plot_ground_truths(truth, [0, 2])
+plotter.fig
+
+# %%
+# Generate and plot detections
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+
+from stonesoup.models.measurement.nonlinear import CartesianToBearingRange
+
+sensor_x = 50  # Placing the sensor off-centre
+sensor_y = 0
+
+measurement_model = CartesianToBearingRange(
+    ndim_state=4,
+    mapping=(0, 2),
+    noise_covar=np.diag([np.radians(0.2), 1]),  # Covariance matrix. 0.2 degree variance in
+    # bearing and 1 metre in range
+    translation_offset=np.array([[sensor_x], [sensor_y]])  # Offset measurements to location of
+    # sensor in cartesian.
+)
+
+# %%
+
+from stonesoup.types.detection import Detection
+
+measurements = []
+for state in truth:
+    measurement = measurement_model.function(state, noise=True)
+    measurements.append(Detection(measurement, timestamp=state.timestamp,
+                                  measurement_model=measurement_model))
+
+# %%
+
+plotter.plot_measurements(measurements, [0, 2])
+plotter.fig
+
+# %%
+# Set up predictor and updater
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# In this EnKF example we must use the :class:`~.EnsemblePredictor`, and choose to use the
+# standard :class:`~.EnsembleUpdater`. Note that we could instanciate any of the other (ensemble)
+# updaters mentioned in this example, in place of the :class:`~.EnsembleUpdater`.
+
+
+from stonesoup.predictor.ensemble import EnsemblePredictor
+from stonesoup.updater.ensemble import EnsembleUpdater
+
+predictor = EnsemblePredictor(transition_model)
+updater = EnsembleUpdater(measurement_model)
+
+
+# %%
+# Prior state
+# ^^^^^^^^^^^
+# For the simulation we must provide a prior state. The Ensemble Filter in stonesoup requires
+# this to be an :class:`~.EnsembleState`. We generate an :class:`~.EnsembleState` by calling
+# :meth:`~.generate_ensemble`. The prior state stores the prior state vector - see below that
+# for the :class:`~.EnsembleState`, the `state_vector` must be a :class:`~.StateVectors` object.
+
+
+from stonesoup.types.state import EnsembleState
+
+ensemble = EnsembleState.generate_ensemble(
+    mean=np.array([[0], [1], [0], [1]]),
+    covar=np.diag([[1.5, 0.5, 1.5, 0.5]]), num_vectors=100)
+prior = EnsembleState(state_vector=ensemble, timestamp=start_time)
+
+# %%
+
+type(prior.state_vector)
+
+
+# %%
+# Run the EnKF
+# ^^^^^^^^^^^^
+
+from stonesoup.types.hypothesis import SingleHypothesis
+from stonesoup.types.track import Track
+
+track = Track()
+for measurement in measurements:
+    prediction = predictor.predict(prior, timestamp=measurement.timestamp)
+    hypothesis = SingleHypothesis(prediction, measurement)  # Group a prediction and measurement
+    post = updater.update(hypothesis)
+    track.append(post)
+    prior = track[-1]
+
+plotter.plot_tracks(track, [0, 2], uncertainty=True, particle=True)
+plotter.fig
```

Comment:
Could you explain what the particles/ellipses represent here? It could be a bit confusing having both plotted otherwise (especially as they are labelled Particles)